File: src/lib/api/system.ts
import { invoke } from "@tauri-apps/api/core";

export const getAutostart = () =>
  invoke<boolean>("get_autostart_enabled");

export const setAutostart = (enabled: boolean) =>
  invoke<void>("set_autostart_enabled", { enabled });

export const backup = () =>
  invoke<string>("backup_database");

export const factoryReset = () =>
  invoke<void>("factory_reset");

File: src-tauri/src/commands/system.rs
use tauri::State;
use crate::error::AppResult;
use crate::services::system as svc;
use crate::state::{DbState, get_conn};

#[tauri::command]
pub async fn get_autostart_enabled(app: tauri::AppHandle) -> bool {
    svc::get_autostart(&app).await
}

#[tauri::command]
pub async fn set_autostart_enabled(app: tauri::AppHandle, enabled: bool) -> AppResult<()> {
    svc::set_autostart(&app, enabled).await

}

#[tauri::command]
pub async fn backup_database() -> AppResult<String> {
    svc::backup_database().await
}

#[tauri::command]
pub async fn factory_reset(state: State<'_, DbState>) -> AppResult<()> {
    let conn = get_conn(&state).await?;
    svc::factory_reset(&conn).await
}
